### DISCUSSION TITLE
Arabic fonts in reports

### DISCUSSION THREAD

**Discussion by yelizariev** [Tecnativa/doodba#533] [Show and tell]
Short notes about making arabic fonts work in doodba template

# Before

![image](https://user-images.githubusercontent.com/186131/213429334-7096224d-c9ad-44f9-ae20-def9c280515d.png)

# After

![image](https://user-images.githubusercontent.com/186131/213429375-3ccf9520-21a0-40f4-9a44-9aae42682200.png)

# How

./odoo/Dockerfile

```
ARG ODOO_VERSION
FROM ghcr.io/tecnativa/doodba:${ODOO_VERSION}-onbuild
ONBUILD ARG FONT_MONO="Noto Mono"
ONBUILD ARG FONT_SANS="Noto Sans"
ONBUILD ARG FONT_SERIF="Noto Serif"
```

./odoo/custom/dependencies/apt.txt

```
fonts-noto-core
```

**Comment by pedrobaeza** [comment]
Is this something that can be added to Doodba for both arabic/non arabic ? Or maybe something to be put to Doodba Copier Template? Why 2 discussions on the same topic?

**Reply by yelizariev** [reply]
I can't find previous discussion. Do you have a link?

Default system fonts is Liberation fonts [1], which don't support arabic fonts apparently. Maybe switch it to Noto fonts  ?

![image](https://user-images.githubusercontent.com/186131/214001408-fd7b8b4b-7fa3-41b2-923a-ff0ad7e334b5.png)


[1]
https://github.com/Tecnativa/doodba/blob/2c6a70b6e16ed6cbc1ba4e35984f6d8339ed1785/16.0.Dockerfile#L45
https://github.com/Tecnativa/doodba/blob/2c6a70b6e16ed6cbc1ba4e35984f6d8339ed1785/16.0.Dockerfile#L184-L186

**Reply by ap-wtioit** [reply]
I think Pedro means that you (accidentally?) posted the discussion twice #533 and #534.

**Reply by yelizariev** [reply]
I see. The second one should be cancelled of course

**Reply by pedrobaeza** [reply]
Are they totally replaceable? Can you install them from the distribution?

**Reply by yelizariev** [reply]
If understand the second question correctly, yes it can be installed from the distribution:
`
apt install fonts-noto-core
`

As for first question, I don't really know what to compare except the support of different languages.

**Reply by giarve** [reply]
Wouldn't it be better to use noto core fonts it by default? What are the downsides?

It applies to all odoo instances with arabic speaking clients, otherwise this hack needs to be done all the times.

**Reply by pedrobaeza** [reply]
We are not sure if they are totally replaceable. Do you know it?

**Reply by giarve** [reply]
> We are not sure if they are totally replaceable. Do you know it?

So far I did not find any issues performance-wise. In terms of reading quality, there is not a big difference, it just supports more unicode characters and when installing the whole package from apt it will take a few more MBs to download in comparison with liberation fonts.

Users can still change font in reports, it is just that characters that could not be displayed they will be able to, now. Also, from what I have seen Noto is one of the most comprehensive fonts **for most languages** (chinese, korean, etc.) in the world.


Selected **oswald** font in reports config in settings:
![image](https://github.com/user-attachments/assets/c8685060-9ed1-4945-9be5-107482c8904f)

see that the arabic chars are still rendered under "San Francisco" in the contact widget but applying the oswald for to the rest of the report (no breaking change for wkhtmltopdf).

**Reply by pedrobaeza** [reply]
@josep-tecnativa something to be considered.
